controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -16,92 +16,110 @@
 from controller.tests import utils
 
 
-class AbstractControllerTest(unittest.TestCase):
+class BaseControllerSetUp(unittest.TestCase):
 
     def setUp(self):
         patch("controller.array_action.array_mediator_xiv.XIVArrayMediator._connect").start()
         detect_array_type_patcher = patch("controller.controller_server.csi_controller_server.detect_array_type")
         self.detect_array_type = detect_array_type_patcher.start()
         self.detect_array_type.return_value = "a9k"
         self.addCleanup(detect_array_type_patcher.stop)
+        self.fqdn = "fqdn"
+        self.mediator = XIVArrayMediator("user", "password", self.fqdn)
+        self.mediator.client = Mock()
+
+        self.storage_agent = MagicMock()
+        self.storage_agent.get_mediator.return_value.__enter__.return_value = self.mediator
+
+        self.servicer = ControllerServicer(self.fqdn)
+
+        self.request = Mock()
+
+
+class CommonControllerTest:
 
     @abc.abstractmethod
-    def get_create_object_method(self):
+    def get_tested_method(self):
         raise NotImplementedError
 
     @abc.abstractmethod
-    def get_create_object_response_method(self):
+    def get_tested_method_response_class(self):
         raise NotImplementedError
 
-    def _test_create_object_with_empty_name(self, storage_agent):
+    def _test_request_with_empty_name(self, storage_agent):

```

Comment:
why the change here? is this used by non-create-object unit tests?

Reply:
it can be

Reply:
I can't think of any. could you provide an example?

Reply:
no

Reply:
then please undo this change

Reply:
done